services/blockchain-indexer/shared/dataService/business/transactions.js

```diff
@@ -233,7 +234,15 @@ const getTransactionsByBlockID = async blockID => {
 				{ id: transaction.id, limit: 1 },
 				['executionStatus'],
 			);
-			transaction.executionStatus = indexedTxInfo.executionStatus;
+
+			if (!indexedTxInfo.executionStatus) {
+				const events = await getEventsByHeight(block.height);
+				transaction.executionStatus = indexedTxInfo.executionStatus
+					? indexedTxInfo.executionStatus
+					: await getTransactionExecutionStatus(transaction, events);
```

Comment:
Since we will come here only when `!indexedTxInfo.executionStatus` is satisfied. we can directly assign the value.

```
				transaction.executionStatus = await getTransactionExecutionStatus(transaction, events);
```